### ISSUE TITLE
Intellisense not detected current namespace.

### ISSUE THREAD

**Issue by dhicksNTIA** [microsoft/vscode-cpptools#1029]
* Operating System and version
    * Linux Mint 18.1
* VS Code version and if you are using the Insiders build
    * 1.16.0
* C/C++ extension version
    * 0.12.4

The issue that I am experiencing is with namespace scoping on Intellisense. Consider this header and source file:

test.h
```cpp
#ifndef TEST_H_
#define TEST_H_

namespace some_space {

enum class SomeEnum { one, two, three };

auto print_enum(SomeEnum some_enum) -> void;

}  // namespace some_space

#endif  // TEST_H_
```

test.cpp
```cpp
#include <iostream>

#include "test.h"

namespace some_space {

auto print_enum(SomeEnum some_enum) -> void {
    switch (some_enum) {
        case SomeEnum::one:
            std::cout << "Enum value is one!" << std::endl;
            break;
        case SomeEnum::two:
            std::cout << "Enum value is two!" << std::endl;
            break;
        case SomeEnum::three:
            std::cout << "Enum value is three!" << std::endl;
            break;
    }
}

}  // namespace some_space
```

When writing lines like `case SomeEnum::three` the Intellisense will not detect the member `three`.
However, if I change those lines to be `case some_space::SomeEnum::three` then the Intellisense will detect the member `three`. With that in mind, I shouldn't have to write out `case some_space::SomeEnum::three` because I am already working within the namespace `some_space`; which is why I think the problem is that the Intellisense is not detecting the current active namespaces.

**Comment by jophippe** [comment]
Hmm, I can't repro this with your test files on Ubuntu, and it seems unlikely to me that this a Mint specific issue. Are you sure Intellisense is running? I can repro this if I turn Intellisense off. Double check that hover and error squiggles work, and the red flame icon on the bottom right corner is not there.

**Comment by sean-mcmanus** [comment]
Yeah, I believe he's describing a known bug with the Tag Parser. https://github.com/Microsoft/vscode-cpptools/issues/721 is another example. Switching the intelliSenseEngine to "Default" is the recommended workaround for now.

**Comment by sean-mcmanus** [comment]
This bug appears fixed with the latest 0.14.6 (I think we fixed it a while a go). Please re-activate if you're still reproing it.
